lfs.h

```diff
@@ -123,12 +123,14 @@ enum lfs_type {
 enum lfs_open_flags {
     // open flags
     LFS_O_RDONLY = 1,         // Open a file as read only
+#ifndef LFS_READONLY
     LFS_O_WRONLY = 2,         // Open a file as write only
     LFS_O_RDWR   = 3,         // Open a file as read and write
     LFS_O_CREAT  = 0x0100,    // Create a file if it does not exist
     LFS_O_EXCL   = 0x0200,    // Fail if a file already exists
     LFS_O_TRUNC  = 0x0400,    // Truncate the existing file to zero size
     LFS_O_APPEND = 0x0800,    // Move to end of file on every write
+#endif
 
     // internally used flags
     LFS_F_DIRTY   = 0x010000, // File does not match storage
```

Comment:
Is it possible (and worth it) also #if'ing these constants like `LFS_F_DIRTY` which are not supposed to be used in read-only mode?

Reply:
It was easy for `LFS_F_DIRTY` and `LFS_F_WRITING`.

However, I am not sure about `LFS_F_ERRED`.
It's documented as "An error occurred during write", but it has one non-obvious use at the end of `int lfs_file_opencfg()`.
It seems you could end up [there](https://github.com/maximevince/littlefs/blob/83fd2e868855bacd3e6e8ae95876af3c1895580a/lfs.c#L2594.) when e.g. `lfs_malloc()` [fails](https://github.com/maximevince/littlefs/blob/83fd2e868855bacd3e6e8ae95876af3c1895580a/lfs.c#L2559), or when opening a file for read hits an `LFS_ERR_NOENT` ([here](https://github.com/maximevince/littlefs/blob/83fd2e868855bacd3e6e8ae95876af3c1895580a/lfs.c#L2535)) (among other possibilities). 
Both of which cases do not seem to be strictly write-related.
So I conclude that this constant is needed in `LFS_READONLY` mode as well.

This is implemented in 35cc31e

Reply:
Ah, it's not actually used in `lfs_file_close` but in the implicit `lfs_file_sync`. The purpose of the LFS_F_ERRED flag is to tell `lfs_file_close` to release resources without writing out to disk. `lfs_file_opencfg` immediately calls `lfs_file_close` if an error occurs.

You have the implicit `lfs_file_sync` removed in LFS_READONLY mode, so it should be safe to remove [this line](https://github.com/maximevince/littlefs/blob/83fd2e868855bacd3e6e8ae95876af3c1895580a/lfs.c#L2594) as well.

Reply:
I only noticed this comment now. I have `ifdef`ed `LFS_F_ERRED` out completely. (99c960b)